plan/physical_plan_builder.go

```diff
@@ -325,6 +323,50 @@ func addPlanToResponse(p PhysicalPlan, planInfo *physicalPlanInfo) *physicalPlan
 	return &physicalPlanInfo{p: np, cost: planInfo.cost, count: planInfo.count}
 }
 
+func enforceProperty(prop *requiredProperty, info *physicalPlanInfo) *physicalPlanInfo {
+	if len(prop.props) != 0 {
+		items := make([]*ByItems, 0, len(prop.props))
+		for _, col := range prop.props {
+			items = append(items, &ByItems{Expr: col.col, Desc: col.desc})
+		}
+		sort := &Sort{
+			ByItems:   items,
+			ExecLimit: prop.limit,
+		}
+		sort.SetSchema(info.p.GetSchema())
+		info = addPlanToResponse(sort, info)
+		count := info.count
+		if prop.limit != nil {
+			count = prop.limit.Offset + prop.limit.Count
+		}
+		info.cost += float64(info.count)*cpuFactor + float64(count)*memoryFactor
+	} else if prop.limit != nil {
+		limit := prop.limit.Copy()
+		limit.SetSchema(info.p.GetSchema())
+		info = addPlanToResponse(limit, info)
+	}
+	if prop.limit != nil && prop.limit.Count < info.count {
+		info.count = prop.limit.Count
+	}
+	return info
+}
+
+func removeLimit(prop *requiredProperty, removeAll bool) *requiredProperty {
```

Comment:
Add comments about when and why we need to remove limit, and when removeAll need to be false when removeAll need to be true.